shotover-proxy/src/message/mod.rs

```diff
@@ -178,18 +178,20 @@ impl Message {
         }
     }
 
-    pub fn message_count(&self) -> Result<NonZeroU32> {
+    /// Get the count of "requests" in this message. In Shotover, all messages are only 1 request except for batch messages which can contain many requests. If a batch message has
+    /// no inner reqeusts, it's count is 1.
```

Comment:
reqeusts -> requests
I think the wording "1 or more requests" gives more confidence to the possible values than "many requests"
a message can either be a response or a request but a response still contains 1 request? :thinking:

Reply:
Hmm, I'm thinking we frame it more in the context of rate limiting then. Since the terminology I tried to use here is too confusing. Perhaps "cell", like what the GCRA/library call it. 

https://docs.rs/governor/latest/governor/struct.RateLimiter.html#method.check

Reply:
Replacing "request" with "cell" could work.
Maybe something like "submessage" would be more self explanatory though. e.g. `message.submessage_count()`

Reply:
To me that is a bit confusing, since BATCH messages are the only ones that actually have "submessages".

Reply:
well at least with submessage its a word that doesnt already mean anything.
So we can define a meaning here and it means exactly what we say it means.
But message and request already have preexisting meanings in shotover.

I also dont see how that differs to cell, only BATCH messages can contain cells.

But I can see an argument for making this method designed around getting values useful for rate limiting and documenting it as such.
Update the docs with your cell approach and we'll see how it reads.

Reply:
Cell means "ticket" in the rate limiter. So most messages have one "ticket" that needs to get through the rate limiter, whereas BATCH messages have many. It's less about the actual content of the message and more about what the rate limiter considers them to be.